### PR TITLE
[Documentation][Configure scripts] Adding documentation to compile using MinGW in Windows

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR adds the documentation needed to install and compile on Windows using *MinGW*. Also updates the corresponding scripts and improves in general the current install documentation.

**NOTE:** Unfortunately, due to a recent bug it is possible to compile on Windows only with GCC,[ but not LLVM/Clang](https://github.com/llvm/llvm-project/issues/53433).

**🆕 Changelog**

- [Update INSTALL.md](https://github.com/KratosMultiphysics/Kratos/commit/d42186c914cef452e9ed8e8c4f26e3bce8d2b9b7)
- [GCC by default but can be manually set](https://github.com/KratosMultiphysics/Kratos/commit/297371e05cf5040b568ca194f2a4957c8753f875)
- [Update MinGW configures](https://github.com/KratosMultiphysics/Kratos/commit/9fa5b7f9d277bfe0fba4fb0278e0e62605e29681)
- [Extend MinGW doc](https://github.com/KratosMultiphysics/Kratos/commit/2f32de979db516429d1efd9eb4ccc77ff07b2ac1)
- [Update banner](https://github.com/KratosMultiphysics/Kratos/commit/a1612e85795632b9ffa98136d21454424084661d)
- [Update style](https://github.com/KratosMultiphysics/Kratos/commit/7ec83002ff759b96e1476e7b1330b11ef4794834)

### CODE REVIEW

**Comment by loumalouomega** [comment]
> **📝 Description**
> 
> This PR adds the documentation needed to install and compile on Windows using _MinGW_. Also updates the corresponding scripts and improves in general the current install documentation.
> 
> **NOTE:** Unfortunately, due to a recent bug it is possible to compile in Windows only with GCC, but not LLVM/Clang.
> 
> **🆕 Changelog**
> 
>     * [Update INSTALL.md](https://github.com/KratosMultiphysics/Kratos/commit/d42186c914cef452e9ed8e8c4f26e3bce8d2b9b7)
> 
>     * [GCC by default but can be manually set](https://github.com/KratosMultiphysics/Kratos/commit/297371e05cf5040b568ca194f2a4957c8753f875)
> 
>     * [Update MinGW configures](https://github.com/KratosMultiphysics/Kratos/commit/9fa5b7f9d277bfe0fba4fb0278e0e62605e29681)
> 
>     * [Extend MinGW doc](https://github.com/KratosMultiphysics/Kratos/commit/2f32de979db516429d1efd9eb4ccc77ff07b2ac1)
> 
>     * [Update banner](https://github.com/KratosMultiphysics/Kratos/commit/a1612e85795632b9ffa98136d21454424084661d)
> 
>     * [Update style](https://github.com/KratosMultiphysics/Kratos/commit/7ec83002ff759b96e1476e7b1330b11ef4794834)

@roigcarlo I replaced Linux with GNU/Linux https://en.wikipedia.org/wiki/GNU/Linux_naming_controversy

Otherwise Stallman-san will be angry

image

**Comment by matekelemen** [comment]
> I replaced Linux with GNU/Linux

you mean GNU/systemd/Linux?

File: INSTALL.md [hunk 1]
```diff
@@ -189,31 +252,38 @@ rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeCache.txt"
 rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeFiles"
 
 # Configure
-cmake -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}" -DUSE_MPI=OFF -DUSE_EIGEN_MKL=OFF
+cmake -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}" \
+-DUSE_MPI=OFF                                                       \
+-DUSE_EIGEN_MKL=OFF                                                 \
+-DKRATOS_GENERATE_PYTHON_STUBS=ON                                   \
```
Comment (roigcarlo) [inline INSTALL.md]: Trailing \ could give problems.

File: INSTALL.md [hunk 2]
```diff
@@ -163,18 +226,18 @@ add_app () {
 }
 
 # Set compiler
-export CC=gcc
-export CXX=g++
+export CC=${CC:-gcc}
+export CXX=${CXX:-g++}
```
Comment (roigcarlo) [inline INSTALL.md]: I would not put this into the example one. As a matter of fact I would remove it from the `KRATOS_SOURCE` as well.

This syntax is meant only for compiling in env where is more convenient to define envs outside the script, but makes no sense for people if they copy paste it

File: INSTALL.md [hunk 3]
```diff
@@ -163,18 +226,18 @@ add_app () {
...
 export KRATOS_BUILD="${KRATOS_SOURCE}/build"
 export KRATOS_APP_DIR="${KRATOS_SOURCE}/applications"
-export KRATOS_INSTALL_PYTHON_USING_LINKS=ON
+# export KRATOS_INSTALL_PYTHON_USING_LINKS=ON
 
 # Set basic configuration
-export KRATOS_BUILD_TYPE="Release"
-export PYTHON_EXECUTABLE="/usr/bin/python3"
+export KRATOS_BUILD_TYPE=${KRATOS_BUILD_TYPE:-"Release"}
+export PYTHON_EXECUTABLE=${PYTHON_EXECUTABLE:-"/usr/bin/python3"}
```
Comment (roigcarlo) [inline INSTALL.md]: Same

File: INSTALL.md [hunk 4]
```diff
@@ -229,9 +299,10 @@ rem set KRATOS_PARALLEL_BUILD_FLAG=/MP4
 
 rem Configure
 @echo on
-cmake -G"Visual Studio 16 2019" -H"%KRATOS_SOURCE%" -B"%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%"  ^
--DUSE_EIGEN_MKL=OFF        ^
--DCMAKE_CXX_FLAGS=" %KRATOS_PARALLEL_BUILD_FLAG% "
+cmake -G"Visual Studio 16 2019" -H"%KRATOS_SOURCE%" -B"%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%"          ^
+-DUSE_EIGEN_MKL=OFF                                                                                 ^
+-DCMAKE_CXX_FLAGS=" %KRATOS_PARALLEL_BUILD_FLAG% "                                                  ^
+-DKRATOS_GENERATE_PYTHON_STUBS=ON                                                                   ^
```
Comment (roigcarlo) [inline INSTALL.md]: Same with ^. I think we have to change it across the repo...

**Review by roigcarlo — COMMENTED** [review]
:+1: Just a couple of comments

File: INSTALL.md (hunk 1 above)
Comment (loumalouomega) [reply INSTALL.md]: In Windows using shell command not

File: INSTALL.md (hunk 2 above)
Comment (loumalouomega) [reply INSTALL.md]: Okay, but literally we are copying the scripts

File: INSTALL.md (hunk 3 above)
Comment (loumalouomega) [reply INSTALL.md]: Okay, As it is doc i guess is no problem

File: INSTALL.md (hunk 4 above)
Comment (loumalouomega) [reply INSTALL.md]: I thing we are mixing things

File: INSTALL.md (hunk 1 above)
Comment (loumalouomega) [reply INSTALL.md]: Okay I see you mean in the Install.md

File: INSTALL.md (hunk 4 above)
Comment (roigcarlo) [reply INSTALL.md]: Yep sorry, but I saw it here. I will make a pr fixing this.

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**